### DISCUSSION TITLE
[Swift SDK] The pairingTopic property is deprecated.

### DISCUSSION THREAD

**Discussion by 0xh3rman** [WalletConnect/walletconnect-monorepo#5390] [Developer Support]
```swift
public struct Session: Codable {
    public let topic: String
    @available(*, deprecated, message: "The pairingTopic property is deprecated.")
    public let pairingTopic: String
    public let peer: AppMetadata
    public let requiredNamespaces: [String: ProposalNamespace]
    public let namespaces: [String: SessionNamespace]
    public let sessionProperties: [String: String]?
    public let expiryDate: Date
    public static var defaultTimeToLive: Int64 {
        WCSession.defaultTimeToLive
    }
}
```

Do you have any migration guide for this warning? what is the equivalent of pairing topic in future release?

**Comment by simonmcl** [comment]
same issue here. I had to update version to solve a privacy manaifest issue and now i'm receiving this. I'm trying to display a list of pairings the user has open, but much of the information needed for the UI is inside the session. So I get all the pairings, and for each pairing I search for the active sessions for each. The current deprecation message doesn't mention any solution

**Comment by llbartekll** [comment]
@0xh3rman can you describe how you've been using this feature?

**Reply by 0xh3rman** [reply]
Similar to @simonmcl, we keep track of lists of pairings for each wallet and show them on UI, we have our local session kv store using topic and pairing topic as key to quick get other meta data stuff.

BTW, I also made a fix for the migration guide here: https://github.com/reown-com/reown-docs/pull/75#issuecomment-2375841231

**Reply by llbartekll** [reply]
@0xh3rman it's been never intended for a wallet to list pairings, you should be listing sessions. Session object has a peer property with dapp's metadata, you should be using this instead.

**Reply by 0xh3rman** [reply]
If I understand correctly, pairing topic is one time thingy? if paired, we only need session topic? received `Request.topic` is same as the session topic

**Reply by 0xh3rman** [reply]
@llbartekll Pairing topic is linked with our local wallet id when handles session proposal (the only id in a wc uri), we link the session topic with same wallet id after approval (in received sessions, we look for session topic using the same pairing topic)

Also consider we can connect same dapp on different devices Desktop / Mobile etc, it seems dapp url is not unique enough to fulfill this  purpose

**Reply by llbartekll** [reply]
> If I understand correctly, pairing topic is one time thingy? if paired, we only need session topic? received Request.topic is same as the session topic

correct

>Also consider we can connect same dapp on different devices Desktop / Mobile etc, it seems dapp url is not unique enough to fulfill this purpose

but session topic will be be unique for each device tooo

**Reply by 0xh3rman** [reply]
How do you handle multi-wallet case? if pairing topic is removed from received session, we need to know which wallet initiates this session, or am I missing something?

**Reply by llbartekll** [reply]
Im not sure I understand but there should be nothing included in the pairing that session does not give you.
>we need to know which wallet initiates this session

you mean wallet or dapp? session contains peer's metadata if this is what you need

>How do you handle multi-wallet case?

what do you mean by that?

**Reply by 0xh3rman** [reply]
I mean the wallet side, user can have multi wallet or different addresses inside our mobile wallet app

**Reply by llbartekll** [reply]
that's fine, you can create a session for multiple addresses at once when you approve the session.
take a look at our docs: https://docs.reown.com/walletkit/ios/usage#namespace-builder
